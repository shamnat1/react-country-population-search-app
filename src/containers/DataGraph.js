import React, { useState, useEffect } from "react";
import {useSelector} from 'react-redux';
import {Card} from'react-bootstrap'
import { ResponsiveContainer,BarChart, Bar, XAxis, YAxis,Tooltip,Cell } from "recharts";
const DataGraph = ()=>{
    const country_population = useSelector((state) => state.allCountries.country_population)
    if(country_population && country_population.length === 0)
        return null;
    return (
        <Card className="mt-4 mb-4">
            <Card.Body>
                <Card.Title>Population Graph</Card.Title>
                <ResponsiveContainer  width="100%" height={400} >
                    <BarChart margin={{top: 5, right: 5, left: 40, bottom: 5}} data={country_population}>
                        <XAxis dataKey="name"/>
                        <YAxis dataKey="population"/>
                        <Tooltip />
                        <Bar dataKey="population" fill="#8884d8">
                            {
                                country_population.map((entry) => {
                                    return <Cell key={entry.name} fill={'#'+(Math.random()*0xFFFFFF<<0).toString(16)} />;
                                })
                            }
                        </Bar>
                    </BarChart>
                </ResponsiveContainer>
                </Card.Body>
        </Card>
    )
}
export default DataGraph;