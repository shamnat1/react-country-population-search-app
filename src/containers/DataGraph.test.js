import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { shallow, configure} from "enzyme";
import DataGraph from "./DataGraph";
import { Provider } from 'react-redux'
import store from '../redux/store';

describe("Country Population component", () => {

    let wrapper;
    let testProps;
    beforeEach(() => {
        wrapper = shallow(<Provider store={store}><DataGraph/></Provider>);
    });

    it("CountryPopulation component renders correctly", () => {

        expect(wrapper).toMatchSnapshot();
    });



});