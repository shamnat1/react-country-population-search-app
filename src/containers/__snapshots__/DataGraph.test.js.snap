// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Population component CountryPopulation component renders correctly 1`] = `ShallowWrapper {}`;
